### ISSUE TITLE
"New login" mail has wrong timezone

### ISSUE THREAD

**Issue by KnappeGEIL** [bitwarden/server#494]
Hi,

Everytime I log into bitwarden (on-premise) from a new device I get an email saying that the new login happened but the timestamp in the email is UTC and not UTC+2 as my timezone.

I have setup the docker containers (adding TZ env variable and an entrypoint wrapper script for mssql container because the ubuntu 16.04 in the container is missing tzdata package) so they all are showing the correct time when I execute `docker run -it containername date` and I've also checked that mssql's GETDATE() is showing the correct date. 

What exactly is handling the time in these emails and needs to be adjusted?

**Comment by kspearrin** [comment]
This is not currently adjustable. All timestamps will remain in UTC regardless of what you set the local time of the server to. We can look into adding a setting for this, however.

**Comment by KnappeGEIL** [comment]
Thanks for your reply. I think such a setting would be pretty useful. 

It would be also very helpful to have a global setting to set the timezone for all containers. Maybe I've just not found where this can be set globally, yet but at the moment I'm providing the TZ environment varaible in docker-compose.overrride.yml to all the containers.

The container bitwarden-mssql is running on Ubuntu 16.04 and is missing the package tzdata which is needed to set the timezone. So I'm also providing `DEBIAN_FRONTEND: "noninteractive"` to make the installation of tzdata package non-interactive. I'm then adding a shell script to the container and replace the entrypoint with this script.

```yaml
mssql:
    volumes:
      - "./timezone.sh:/timezone.sh"
    environment:
      TZ: "Europe/Berlin"
      DEBIAN_FRONTEND: "noninteractive"
    entrypoint: /timezone.sh
```

timezone.sh

```bash
#!/bin/sh

apt-get update && apt-get install -y tzdata
rm /etc/localtime
ln -snf /usr/share/zoneinfo/$TZ /etc/localtime
dpkg-reconfigure -f noninteractive tzdata
apt-get clean

/entrypoint.sh
```

Is there a way to do this more easily? Having the correct timezone on all containers would be pretty important to have correct timestamps in the log files.

**Comment by ghaisasadvait** [comment]
Has this option been added yet?

**Comment by Mart124** [comment]
> The container bitwarden-mssql is running on Ubuntu 16.04 and is missing the package tzdata

All containers now have `tzdata` installed (see #606) so that you can set your own timezone thanks to the `TZ` environment variable.

**Comment by ghaisasadvait** [comment]
Thank you for replying!
Can you please guide me as to how to set it?

**Comment by Mart124** [comment]
Log into your containers and run the `date` command, you'll see the effect of setting `TZ` in `bwdata/docker/docker-compose.override.yml`.
For example :
```
services:
  mssql:
    environment:
      - TZ=Europe/Berlin
```
However, as said [above](https://github.com/bitwarden/server/issues/494#issuecomment-490714361), mails will remain in UTC timezone.
_(you should be able to close #973)_

**Comment by ghaisasadvait** [comment]
Oh so regardless of seeing the timezone, the email well still be in UTC?

**Comment by Mart124** [comment]
Yes. Proper timezone allows other things such as proper timestamps in logs, proper cron schedules etc...

**Comment by ghaisasadvait** [comment]
Oh I see. Thank you for your reply!

**Comment by djsmith85** [comment]
Closing this as tzdata was added and @Mart124 has provided information on how to configure it.

**Comment by KnappeGEIL** [comment]
@djsmith85 Sure you can close this but the issue is not mainly about tzdata but about "New login mails" not having the correct timestamp and that it's not configurable. Since this seems to be still the case I wouldn't say that this issue is solved.

**Comment by djsmith85** [comment]
@KnappeGEIL Thank you for your report. Based on your comment, I just checked the mail in question and indeed it is still not possible to configure/set the timezone. Re-opening..

**Comment by ghaisasadvait** [comment]
The email are still not coming in the local/server timezone

**Comment by shadowjig** [comment]
Any update on this?

**Comment by georglauterbach** [comment]
@patrick-bitwarden I am currently facing a related issue: when self-hosting, the "Log In Via Device" functionality does not work because the server works with UTC while the client works with localtime. The result is that the client sees the result as "two hours old", because I am in CEST (UTC+2). Definitely a bug, if you ask me.

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by closebot-bw** [comment]
🔒 **Issue Closed Due to Inactivity**

This issue has been automatically closed due to lack of activity for an extended period. We periodically review and close inactive issues to help maintain our issue tracker and focus on current priorities.

**If this issue is still relevant:**
- Please create a new issue with updated information
- Include steps to reproduce the problem if it's a bug report
- Mention if this issue still occurs in the latest version

Thank you for your contribution to this project. Your feedback helps us improve! 🙏
